lesson_10/solid/solid_app/src/main/java/com/codedifferently/lesson10/library/Library.java

```diff
@@ -35,24 +35,33 @@ public String getId() {
   /**
    * Add a book to the library.
    *
-   * @param book The book to add.
+   * @param librarian The librarian adding book.
+   * @param item The item to add.
    */
-  public void addBook(Book book) {
-    this.bookIds.add(book.getId());
-    book.setLibrary(this);
+  public void addMedia(MediaItem item, Librarian librarian) {
+    if (librarian == null) {
+      throw new IllegalArgumentException("Librarian needed to add media item.");
+    }
+
+    this.mediaIds.add(item.getId());
+    item.setLibrary(this);
   }
 
   /**
-   * Remove a book from the library.
+   * Remove media item from the library.
    *
-   * @param book The book to remove.
+   * @param item The item to remove.
+   * @param librarian The librarian removing item.
    */
-  public void removeBook(Book book) throws BookCheckedOutException {
-    if (this.isCheckedOut(book)) {
-      throw new BookCheckedOutException("Cannot remove checked out book.");
+  public void removeMediaItem(MediaItem item, Librarian librarian) throws BookCheckedOutException {
```

Comment:
Should be `ItemCheckedOutException` as part of this change.

Reply:
Missed that! All changed.